OptimizelySDK/Utils/CollectionExtensions.cs

```diff
@@ -43,16 +42,14 @@ public static class CollectionExtensions
 
             if (right == null)
             {
-                return left;
+                return;
             }
 
             foreach (var kvp in right.Where(
                          kvp => !left.ContainsKey(kvp.Key)))
             {
```

Comment:
We can make it more clear with a comment about "left has higher priority when keys conflict".